packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -142,12 +180,23 @@ class AnalyzeCommand extends FlutterCommand {
       return false;
     }
 
+    // Don't run pub if asking for android analysis.
+    if (boolArg('android')) {
+      return false;
+    }
+
     return super.shouldRunPub;
   }
 
   @override
   Future<FlutterCommandResult> runCommand() async {
-    if (boolArg('suggestions')) {
+
+    if (boolArg('android')) {
+      await AndroidAnalyze(
+        argResults!,
```

Comment:
rather than passing the entire argResults object, can you parse these out here and pass the values to the `AndroidAnalyze` class? This would make testing easier.